src/types.cpp

```diff
@@ -53,8 +53,25 @@ std::string typestr(const SizedType &type, bool debug)
     case Type::array:
       return typestr(*type.GetElementTy(), debug) + "[" +
              std::to_string(type.GetNumElements()) + "]";
-    case Type::record:
-      return type.GetName();
+    case Type::record: {
+      if (!type.IsAnonTy())
+        return type.GetName();
+
+      // For anonymous structs/unions, return a string of the
+      // format "struct { field1type,... }"
+      std::string type_name = type.GetName();
+      std::string res = type_name.substr(0, type_name.find_first_of(" "));
+      size_t n = type.GetFieldCount();
+
+      res += " {";
+      for (size_t i = 0; i < n; ++i) {
+        res += typestr(type.GetField(i).type, debug);
+        if (i != n - 1)
+          res += ",";
+      }
```

Comment:
Just a note: we have the `str_join` utility function but you'd need to create a `std::vector` of `typestr` representations so this is probably cleaner.